src/types.ts

```diff
@@ -39,4 +39,5 @@ export interface AmqpConfig {
    exchanges: AmqpExchangeConfig[]
    queues: AmqpQueueConfig[]
    bindings: AmqpBindingConfig[]
+   channel_timeout?: number
```

Comment:
What if the constructor just took in Partial<AMQPConfig> and then we default everything? That would keep your type safety and keep it backwards compatible.

Reply:
I like it, that's a good idea